experiments/example/main.go

```diff
@@ -0,0 +1,129 @@
+	// Each value needs to be a string so some conversion might be necessary
+	var experimentConfiguration map[string]string
+	experimentConfiguration["min_thread_count"] = strconv.Itoa(minThreadCount)
+	experimentConfiguration["max_thread_count"] = strconv.Itoa(maxThreadCount)
+	experimentConfiguration["load_points"] = loadDuration.String()
+	// Saving command line arguments and name of the binary might help with future analysis.
+	experimentConfiguration["command_arguments"] = strings.Join(os.Args[1:], " ")
+	experimentConfiguration["experiment_name"] = os.Args[0]
+	// Finally - sava configuration to Cassandra.
+	err = metadata.RecordMap(experimentConfiguration)
+	errutil.CheckWithContext(err, "Cannot save metadata in Cassandra Metadata Database")
+
+	// Create mutilate Snap session launcher - it will be used to gather metrics about memcached performance.
+	mutilateSnapSession, err := mutilatesession.NewSessionLauncherDefault()
+	errutil.CheckWithContext(err, "Cannot create Mutilate snap session")
+
+	// Instansce of executor.Executor that allows to launch processes locally.
+	executor := executor.NewLocal()
+
+	// Instance of executor.Launcher that uses Executor and amemcached configuration to launch memcached.
+	memcachedLauncher := memcached.New(executor, memcached.DefaultMemcachedConfig())
+
+	// Instance of executor.LoadGenerator that will launch mutilate locally,
+	mutilateLauncher := mutilate.New(executor, mutilate.DefaultMutilateConfig())
+
+	// Iterate over range of cores defined in the experiment configuration.
+	for threadCount := minThreadCount; threadCount <= maxThreadCount; threadCount++ {
+		// Iterate over number of load points.
+		for loadPoint := 1; loadPoint <= numberOfLoadPoints; loadPoint++ {
+			// Calculating number of QPS to be sent to memcached.
+			qps := int(maxQPS / numberOfLoadPoints * loadPoint)
+			// The closure represents single phase of experiment (combination of number QPS and number of threads). It allows to use defer to stop all the processes launched.
+			phase := func() {
+				// Launching memcached...
+				memcached, err := memcachedLauncher.Launch()
+				errutil.CheckWithContext(err, fmt.Sprintf("Cannot launch memcached with %d threads, %d QPS and load duration of %s", threadCount, qps, loadDuration))
+				// Making sure to stop memcached at the end of phase.
+				defer memcached.Stop()
+
+				// Launching mutilate to stress memcached...
+				mutilateTask, err := mutilateLauncher.Load(qps, loadDuration)
+				errutil.CheckWithContext(err, fmt.Sprintf("Cannot launch mutilate with %d threads, %d QPS and load duration of %s", threadCount, qps, loadDuration))
+				defer mutilateTask.Stop()
+				// Waiting until load generating finishes.
+				mutilateTask.Wait(0)
+
+				// Preparing tags for Snap metrics (they will be used to identify unique phases).
+				tags := map[string]interface{}{
+					"thread_count": threadCount,
```

Comment:
add predefined field (constant sensitivity) to store experiment_id (otherwise I cannot load this data from jupyter)